File: examples/governance_static_go/domain/deactivated_stakeholder.go
package domain

import "time"

type DeactivatedStakeholder struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	StakeholderId string `json:"stakeholder_id"`
	Name string `json:"name"`
	Email string `json:"email"`
	Role string `json:"role"`
	Team string `json:"team"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e DeactivatedStakeholder) EventName() string { return "DeactivatedStakeholder" }

func (e DeactivatedStakeholder) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/identity_domain/lib/identity_domain/stakeholder/events/deactivated_stakeholder.rb
module IdentityDomain
  class Stakeholder
    module Events
      class DeactivatedStakeholder
        attr_reader :aggregate_id, :stakeholder_id, :name, :email, :role, :team, :status, :occurred_at

        def initialize(aggregate_id: nil, stakeholder_id: nil, name: nil, email: nil, role: nil, team: nil, status: nil)
          @aggregate_id = aggregate_id
          @stakeholder_id = stakeholder_id
          @name = name
          @email = email
          @role = role
          @team = team
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
